components/role-mgt/org.wso2.carbon.identity.role.mgt.core/src/main/java/org/wso2/carbon/identity/role/mgt/core/RoleManagementService.java

```diff
@@ -0,0 +1,173 @@
+    /**
+     * Delete the given role.
+     *
+     * @param roleID       Role ID.
+     * @param tenantDomain Tenant domain.
+     * @throws IdentityRoleManagementException IdentityRoleManagementException.
+     */
+    void deleteRole(String roleID, String tenantDomain) throws IdentityRoleManagementException;
+
+    /**
+     * Get user list of the given role.
+     *
+     * @param roleID       Role ID.
+     * @param tenantDomain Tenant domain.
+     * @return List of users.
+     * @throws IdentityRoleManagementException IdentityRoleManagementException.
+     */
+    List<UserBasicInfo> getUserListOfRole(String roleID, String tenantDomain) throws IdentityRoleManagementException;
+
+    /**
+     * Update the user list of the given role.
+     *
+     * @param roleID          Role ID.
+     * @param newUserList     The set of new users.
+     * @param deletedUserList The set of deleted users.
+     * @param tenantDomain    Tenant domain.
+     * @throws IdentityRoleManagementException IdentityRoleManagementException.
+     */
+    RoleBasicInfo updateUserListOfRole(String roleID, List<String> newUserList, List<String> deletedUserList,
+            String tenantDomain) throws IdentityRoleManagementException;
+
+    /**
+     * Get group list of the given role.
+     *
+     * @param roleID       Role ID.
+     * @param tenantDomain Tenant domain.
+     * @return List of groups.
+     * @throws IdentityRoleManagementException IdentityRoleManagementException.
+     */
+    List<GroupBasicInfo> getGroupListOfRole(String roleID, String tenantDomain) throws IdentityRoleManagementException;
+
+    /**
+     * Update the group list of the given role.
+     *
+     * @param roleID                Role ID.
+     * @param newGroupNamesList     The set of new groups.
+     * @param deletedGroupNamesList The set of deleted groups.
+     * @param tenantDomain          Tenant domain.
+     * @throws IdentityRoleManagementException IdentityRoleManagementException.
+     */
+    RoleBasicInfo updateGroupListOfRole(String roleID, List<String> newGroupNamesList,
+            List<String> deletedGroupNamesList, String tenantDomain) throws IdentityRoleManagementException;
+
+    /**
+     * Get permission list of the given role.
+     *
+     * @param roleID       Role ID.
+     * @param tenantDomain Tenant domain.
+     * @return List of permissions.
+     * @throws IdentityRoleManagementException IdentityRoleManagementException.
+     */
+    List<String> getPermissionListOfRole(String roleID, String tenantDomain) throws IdentityRoleManagementException;
+
+    /**
+     * Set the list of permission to the given role.
+     *
+     * @param roleID      Role ID.
+     * @param permissions List of permissions.
+     * @throws IdentityRoleManagementException IdentityRoleManagementException.
+     */
+    RoleBasicInfo setPermissionsForRole(String roleID, List<String> permissions, String tenantDomain)
+            throws IdentityRoleManagementException;
+
+    /**
```

Comment:
So this will be a PUT updating full permission list

Reply:
Yes. We have used the available UserRealmProxy for mange permissions which only have the capability to set full list.

Reply:
This is the existing `RolePermissionManagementServiceImpl`. We have moved the same implementation to the role management service. https://github.com/wso2/carbon-identity-framework/blob/master/components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/RolePermissionManagementServiceImpl.java

Reply:
ok